### PR TITLE
rosetta delegator address balances

### PR DESCRIPTION
**Pull request by zizhengjiang**
### Description
Implement the delegator address balances rosetta endpoint.

The rosetta client should manage its own delegator and pool address relationship when fetching relevant data.
The request payload should look like this (added the metadata field under sub_account_identifier):
```
"account_identifier": {
    "address": "0x899d44dca988af7895cdc9828ae0e7f6c5c42d74fa1610e693fb5f613cb8c09a", // owner/delegator address
    "sub_account": {
        "address": "active_stake", // string active
        # this is the new field added
        "metadata": {
            "pool_address": "0xbd50bb20e1972747db365c3a6e31255d78867c6b8df9877ea6054bb335dd6031" // pool_address
        }
    }
}
```

**About balance fetching**:
Using delegator address [0x65b1e4d62e803d7e3a5c715baa33391757ae530c9fef287bf2d7d15c1ea6e3e6](https://explorer.aptoslabs.com/account/0x65b1e4d62e803d7e3a5c715baa33391757ae530c9fef287bf2d7d15c1ea6e3e6?network=mainnet) and its validator address [0x2ded75e99c6efbe143a9648a3e88ebf9d0cf249b2af44d510bdd0287e8adcc79](https://explorer.aptoslabs.com/account/0x2ded75e99c6efbe143a9648a3e88ebf9d0cf249b2af44d510bdd0287e8adcc79?network=mainnet) as an example

The response should be similar to native staking:
Variables that should be fetched from the **delegator address:**
1. Sequence number from its `0x1::account::Account` resource
2. Balances from delegation_pool.move get_stake() function

Variables should be fetched from the **pool address:**
~~1. Operator from `0x1::stake::StakePool` resource~~
2. Lockup until secs from `0x1::stake::StakePool` resource

### Test Plan

Tested with remote online node from e2e
**delegated staking balance**
Screenshot 2023-06-01 at 4 15 45 PM

**native staking balance**
Screenshot 2023-06-01 at 4 16 37 PM

**direct balance**
Screenshot 2023-06-01 at 4 17 25 PM

### CODE REVIEW

File: crates/aptos-rosetta/src/types/identifiers.rs
Comment (xindingw) [inline crates/aptos-rosetta/src/types/identifiers.rs:279]: I understand this is not from this PR but why we have to do this hack? by any means, Neither `STAKE`, `PENDING_INACTIVE_STAKE ` or `INACTIVE_STAKE` is an address. If we just need a enumeration, I believe `metadata` field is for this purpose, right?

File: crates/aptos-rosetta/src/types/misc.rs
Comment (zizhengjiang) [reply crates/aptos-rosetta/src/types/misc.rs:102]: will remove it

File: crates/aptos-rosetta/src/types/misc.rs
Comment (zizhengjiang) [reply crates/aptos-rosetta/src/types/misc.rs:372]: here i'd like to just mimic the `get_stake_balances` func and let the upper layer (account.rs) handle the error

File: crates/aptos-rosetta/src/types/identifiers.rs
Comment (zizhengjiang) [reply crates/aptos-rosetta/src/types/identifiers.rs:279]: I feel we probably don't want to deal with extra fields that's why it was initially implemented like this
given the current code is already in use internally i'd prefer just leave it as it is and use the new field get the target data

File: crates/aptos-rosetta/src/types/misc.rs
Comment (xindingw) [reply crates/aptos-rosetta/src/types/misc.rs:372]: No, this is different from the existing code to handle `requested_balance`(it's also problematic though). `rest_client.view` might return an error and in this case the else branch will be entered. In the else branch, the error value is ignored and `OK` is returned so the upper layer cannot even know there is an error.

File: crates/aptos-rosetta/src/types/identifiers.rs
Comment (xindingw) [reply crates/aptos-rosetta/src/types/identifiers.rs:279]: I won't block this PR for this. But I highly suggest to fix it ASAP because it doesn't make sense at all and the schema will be incompatible after fixing.

File: crates/aptos-rosetta/src/types/misc.rs
Comment (zizhengjiang) [reply crates/aptos-rosetta/src/types/misc.rs:397]: will remove the other two, just left this one
it will be handled by the `account.rs` layer

File: crates/aptos-rosetta/src/types/misc.rs
Comment (xindingw) [inline crates/aptos-rosetta/src/types/misc.rs:369]: This still discards the error from `rest_client.view`. You can try `.await?`, which throws the error if there is one.:
```suggestion
    let balances_response = rest_client.view(
        &ViewRequest {
            function: DELEGATION_POOL_GET_STAKE_FUNCTION.clone(),
            type_arguments: vec![],
            arguments: vec![
                serde_json::Value::String(String::from(
                    pool_address.to_string(),
                )),
                serde_json::Value::String(String::from(
                    owner_address.to_string()
                )),
            ],
        },
        Some(version),
    ).await?;
    let result = balances_response.into_inner();
    ...
```

**Review by xiaoxiaff — APPROVED** [review]

**Review by xiaoxiaff — APPROVED** [review]

**Review by xiaoxiaff — APPROVED** [review]

**Review by nicolas-martin — APPROVED** [review]

File: crates/aptos-rosetta/src/types/misc.rs
Comment (xindingw) [inline crates/aptos-rosetta/src/types/misc.rs:359]: nit: Maybe throw "Unable to construct BalanceResult instance" error here?

**Review by xindingw — APPROVED** [review]
LGTM! Let me know if you need me to find another approver.

**Review by davidiw — APPROVED** [review]

**Pull request merged**
